### PR TITLE
ChDKZ tank fix #2 (Camo)

### LINKED ISSUE #2: Unsynchronized destroyed buildings
Destroyed buildings are not synchronized correctly. AI will think a building is destroyed and shoot through them. One player sees the building destroyed while another does not etc.

Might be fixed by commenting out statSave\loadAccount.sqf line 60:
`["destroyedBuildings"] call fn_LoadStat;`

### LINKED ISSUE #2871: [Bug]: rhs ChDKZ template wrong camo tank due to flag issue
### Describe the bug

the flags on the ChDKZ tanks seem to be awfully persistent and somehow for unknown reason they do not go away when the tank goes away.
This is despite the fact that the event handler that RHS uses should be able to deal with removing them correctly.

This issue was hotfixed by replacing the ChDKZ T-72s with AFRF T-72s meaning that they do not wear the ChDKZ camo.

### How to reproduce

Unknown - investigation needed.
Suspect is garaging functionality.
potentially in combination of a dedicated server (Unable to recreate when direct LAN hosting).
May require multiple clients to recreate.

### Version

3.3.0

### Have you altered the code?

No

### What i have changed

_No response_

### Map

Any

### What server?

Community Test Server

### Time bug occured (Server time/UTC)

_No response_

### Mods

```html
community server RHS modset
```

### Additional context

The tanks the template currently uses needs to be replaced with config-patched ChDKZ tanks, preferably they would get to keep their flags as they look good with them, but that is a wish and not a requirement.

Pending fix #2885, which may depend on #2892

### PR DESCRIPTION
**Pull request by ante185**
## What type of PR is this.
1. [x] Bug
2. [x] Change
3. [ ] Enhancement

### What have you changed and why?
Information:
new classes for ChDKZ tanks, minus the event handler that adds the flag to them - they do not have to use the AFRF tanks anymore.
their 1989 version of the T72 is now painted too.
    

### Please specify which Issue this PR Resolves.
closes #2871

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:
I didn't exactly have a lot to go on when setting this up, except that config patches for the bomb weapons but not a lot of parallels with them.
The tanks are hidden from the editor by default, enable debug to unhide them.
I made them separate classnames instead of overwriting the RHS ones just to explicitly not break anything.

Dependency: issue #2892

### CODE REVIEW

**Comment by jaj22** [comment]
I don't think we've really established how to handle load order dependencies. Issue is that we have to guarantee that the config fixes are being applied after GREF is loaded, but only if GREF is loaded, and so on for each mod that we write fixes for. As #if can't be nested in Arma we'll need some other method than what's already in there for CUP. Gonna try a couple and see which looks least horrible.

**Comment by ante185** [comment]
> I don't think we've really established how to handle load order dependencies. Issue is that we have to guarantee that the config fixes are being applied after GREF is loaded, but only if GREF is loaded, and so on for each mod that we write fixes for. As #if can't be nested in Arma we'll need some other method than what's already in there for CUP. Gonna try a couple and see which looks least horrible.

Sounds good, feel free to push to this PR (I think you can do that, right?) Cause this is about as far as I'm able to contribute.

**Comment by jaj22** [comment]
Well, that seems to work. Not 100% happy about the `PATCHNAME(CUP)` stuff but whatever.

The __has_include wrappers can be replaced by that new CfgPatches thing in 2.14 when that shows up.

**Review by ante185 — COMMENTED** [review]
the changes look fine to me

File: A3A/addons/config_fixes/CfgVehicles.hpp [hunk 1]
```diff
@@ -1,3 +1,4 @@
 class CfgVehicles {
+	#include "CfgVehicles\chdkz_rhs.hpp"
+};
```
Comment (ante185) [inline A3A/addons/config_fixes/CfgVehicles.hpp:3]: This entire file doesn't do anything anymore now does it?

**Comment by ante185** [comment]
Tests: load and launch with and without RHS and/or CUP, open editor, place relevant objects, check config browser.
Tested Vanilla: good - classes aren't added.
Tested RHS: good - fixed tanks get added when RHS is loaded.
Tested CUP: good - doesn't patch anything on its own, doesn't cause issues.

**Comment by jaj22** [comment]
I wrote half of this but I'm gonna mark it as ready anyway.

**Review by Bob-Murphy — APPROVED** [review]
lgtm

**Pull request merged**